app/components/market/submit/submit-market-application.html

```diff
@@ -1,37 +1,223 @@
+                        </div>
+                        <div *ngSwitchDefault>
+                            <md-input-container class="form-element">
+                            <input mdInput type="text" name="name" [formControlName]="param" [placeholder]="param">
+                                <md-hint align="end">{{jobTemplate.parameters[param].metadata.description}}</md-hint>
+                            </md-input-container>
+                        </div>
                     </div>
-                    <div *ngSwitchCase="NcjParameterExtendedType.fileInFileGroup">
-                        <bl-cloud-file-picker [containerId]="getContainerFromFileGroup(jobFormGroup.controls[param.dependsOn]?.value)" [formControlName]="param.id"
-                            [label]="param.name" class="form-element" [hint]="param.description">
-                        </bl-cloud-file-picker>
+                </bl-form-section>
+
+            </bl-form-page>
+        </bl-complex-form>
+    </div>
+
+    <div *ngIf="mode_state == 'PoolOJob'">
+        <bl-complex-form [formGroup]="form" [hideFooter]="true">
+            <bl-form-page main-form-page [title]="title">
+                <bl-form-section title="Mode">
+                    <ul>
+                        <button style="margin:5px; width:200px" (click) = "mode_state = 'PoolNJob'" type="button" md-raised-button>new Pool + Job</button>
+                        <button style="margin:5px; width:200px; background-color:#3e81b0;" (click) = "mode_state = 'PoolOJob'" type="button" md-raised-button>existing Pool + Job</button>
+                        <button style="margin:5px; width:200px" (click) = "mode_state = 'Pool'" type="button" md-raised-button>new Pool</button>
+                    </ul>
+                </bl-form-section>
+                <bl-form-section title="Pool">
+                    <bl-pool-picker [formControl]="pickedPool"></bl-pool-picker>
+                </bl-form-section>
+                <bl-form-section title="Job">
+                    <div *ngFor="let param of getParameters(jobTemplate)" [ngSwitch]="getType(param)">
+                        <div *ngSwitchCase="'string'">
+                            <md-input-container class="form-element">
+                            <input mdInput type="text" name="name" [formControlName]="param" [placeholder]="param">
+                                <md-hint align="end">{{jobTemplate.parameters[param].metadata.description}}</md-hint>
+                            </md-input-container>
+                        </div>
+                        <div *ngSwitchCase="'file-group'">
+                            <bl-file-group-picker [formControlName]="param" [label]="param" class="form-element" [hint]="jobTemplate.parameters[param].metadata.description">
+                            </bl-file-group-picker>
+                        </div>
+                        <div *ngSwitchCase="'file-in-file-group'">
```

Comment:
There is way to much duplicates here. You should probably create another component called `ncj-form-parameter` to which you give the param type and it will choose the right value